### ISSUE TITLE
Latest Version Instantly Crashes

### ISSUE THREAD

**Issue by TripleOh7** [AndInTheClouds/chordreader2#51]
The latest version of this app (installed by F-Droid) instantly crashes upon open. I am on the latest Android 14 so F-Droid supports automatically updating apps.

I'm not sure if the multidex support or F-Droid's auto update is making it crash but I'd really appreciate it if you could look at it; I use this app nearly every day and am willing to test, etc.

Thank you for working on such a cool project! I show this to all the other musicians in my life knowing full well they can't do the same on their iPhones :)

https://github.com/user-attachments/assets/50f91624-8a9e-4fe7-99cc-7076a3d14758

**Comment by TripleOh7** [comment]
Did manage to rollback the app to the March release so I can still use that version in the meantime

**Comment by ElizabethWega** [comment]
The same happens to me. I'm using the app on a Xiaomi device running custom EU MIUI ROM. Here's the error log that the built-in phone's program collected:
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.hollowbamboo.chordreader2/org.hollowbamboo.chordreader2.MainActivity}: android.view.InflateException: Binary XML file line #18 in org.hollowbamboo.chordreader2:layout/activity_main: Binary XML file line #17 in org.hollowbamboo.chordreader2:layout/content_main: Error inflating class fragment
	at android.app.ActivityThread.performLaunchActivity(Unknown Source:902)
	at android.app.ActivityThread.handleLaunchActivity(Unknown Source:77)
	at android.app.servertransaction.LaunchActivityItem.execute(Unknown Source:79)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(Unknown Source:77)
	at android.app.servertransaction.TransactionExecutor.execute(Unknown Source:76)
	at android.app.ActivityThread$H.handleMessage(Unknown Source:172)
	at android.os.Handler.dispatchMessage(Unknown Source:19)
	at android.os.Looper.loopOnce(Unknown Source:182)
	at android.os.Looper.loop(Unknown Source:82)
	at android.app.ActivityThread.main(Unknown Source:123)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(Unknown Source:11)
	at com.android.internal.os.ZygoteInit.main(Unknown Source:312)
Caused by: android.view.InflateException: Binary XML file line #18 in org.hollowbamboo.chordreader2:layout/activity_main: Binary XML file line #17 in org.hollowbamboo.chordreader2:layout/content_main: Error inflating class fragment
Caused by: android.view.InflateException: Binary XML file line #17 in org.hollowbamboo.chordreader2:layout/content_main: Error inflating class fragment
Caused by: java.lang.RuntimeException: Exception inflating org.hollowbamboo.chordreader2:navigation/mobile_navigation line 126
	at x0.t.b(SourceFile:1)
	at x0.j.o0(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment$b.g(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment$b.a(SourceFile:1)
	at d2.l.getValue(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment.a2(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment.C0(SourceFile:1)
	at androidx.fragment.app.i.h1(SourceFile:1)
	at androidx.fragment.app.v.e(SourceFile:1)
	at androidx.fragment.app.v.m(SourceFile:1)
	at androidx.fragment.app.o.onCreateView(SourceFile:1)
	at androidx.fragment.app.l.n(SourceFile:1)
	at androidx.fragment.app.j.W(SourceFile:1)
	at androidx.fragment.app.j.onCreateView(SourceFile:1)
	at android.view.LayoutInflater.tryCreateView(Unknown Source:39)
	at android.view.LayoutInflater.createViewFromTag(Unknown Source:42)
	at android.view.LayoutInflater.createViewFromTag(Unknown Source:6)
	at android.view.LayoutInflater.rInflate(Unknown Source:92)
	at android.view.LayoutInflater.rInflateChildren(Unknown Source:9)
	at android.view.LayoutInflater.parseInclude(Unknown Source:302)
	at android.view.LayoutInflater.rInflate(Unknown Source:71)
	at android.view.LayoutInflater.rInflateChildren(Unknown Source:9)
	at android.view.LayoutInflater.parseInclude(Unknown Source:302)
	at android.view.LayoutInflater.rInflate(Unknown Source:71)
	at android.view.LayoutInflater.rInflateChildren(Unknown Source:9)
	at android.view.LayoutInflater.inflate(Unknown Source:112)
	at android.view.LayoutInflater.inflate(Unknown Source:29)
	at androidx.databinding.f.e(SourceFile:1)
	at androidx.databinding.g.m(SourceFile:1)
	at g3.b.v(SourceFile:1)
	at g3.b.u(SourceFile:1)
	at org.hollowbamboo.chordreader2.MainActivity.onCreate(Unknown Source:7)
	at android.app.Activity.performCreate(Unknown Source:127)
	at android.app.Activity.performCreate(Unknown Source:1)
	at android.app.Instrumentation.callActivityOnCreate(Unknown Source:3)
	at android.app.ActivityThread.performLaunchActivity(Unknown Source:627)
	at android.app.ActivityThread.handleLaunchActivity(Unknown Source:77)
	at android.app.servertransaction.LaunchActivityItem.execute(Unknown Source:79)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(Unknown Source:77)
	at android.app.servertransaction.TransactionExecutor.execute(Unknown Source:76)
	at android.app.ActivityThread$H.handleMessage(Unknown Source:172)
	at android.os.Handler.dispatchMessage(Unknown Source:19)
	at android.os.Looper.loopOnce(Unknown Source:182)
	at android.os.Looper.loop(Unknown Source:82)
	at android.app.ActivityThread.main(Unknown Source:123)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(Unknown Source:11)
	at com.android.internal.os.ZygoteInit.main(Unknown Source:312)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: org.hollowbamboo.chordreader2.chords.Chord
	at x0.x$l.a(SourceFile:1)
	at x0.t.d(SourceFile:1)
	at x0.t.f(SourceFile:1)
	at x0.t.a(SourceFile:1)
	at x0.t.a(SourceFile:1)
	... 48 more
Caused by: java.lang.ClassNotFoundException: org.hollowbamboo.chordreader2.chords.Chord
	at java.lang.Class.classForName(Native Method)
	at java.lang.Class.forName(Class.java:454)
	at java.lang.Class.forName(Class.java:379)
	... 53 more
Caused by: java.lang.ClassNotFoundException: org.hollowbamboo.chordreader2.chords.Chord
	... 56 more

```

Here's what I got from LogFox:
```
FATAL EXCEPTION: main
Process: org.hollowbamboo.chordreader2, PID: 26453
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.hollowbamboo.chordreader2/org.hollowbamboo.chordreader2.MainActivity}: android.view.InflateException: Binary XML file line #18 in org.hollowbamboo.chordreader2:layout/activity_main: Binary XML file line #17 in org.hollowbamboo.chordreader2:layout/content_main: Error inflating class fragment
	at android.app.ActivityThread.performLaunchActivity(Unknown Source:902)
	at android.app.ActivityThread.handleLaunchActivity(Unknown Source:77)
	at android.app.servertransaction.LaunchActivityItem.execute(Unknown Source:79)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(Unknown Source:77)
	at android.app.servertransaction.TransactionExecutor.execute(Unknown Source:76)
	at android.app.ActivityThread$H.handleMessage(Unknown Source:172)
	at android.os.Handler.dispatchMessage(Unknown Source:19)
	at android.os.Looper.loopOnce(Unknown Source:182)
	at android.os.Looper.loop(Unknown Source:82)
	at android.app.ActivityThread.main(Unknown Source:123)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(Unknown Source:11)
	at com.android.internal.os.ZygoteInit.main(Unknown Source:312)
Caused by: android.view.InflateException: Binary XML file line #18 in org.hollowbamboo.chordreader2:layout/activity_main: Binary XML file line #17 in org.hollowbamboo.chordreader2:layout/content_main: Error inflating class fragment
Caused by: android.view.InflateException: Binary XML file line #17 in org.hollowbamboo.chordreader2:layout/content_main: Error inflating class fragment
Caused by: java.lang.RuntimeException: Exception inflating org.hollowbamboo.chordreader2:navigation/mobile_navigation line 126
	at x0.t.b(SourceFile:1)
	at x0.j.o0(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment$b.g(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment$b.a(SourceFile:1)
	at d2.l.getValue(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment.a2(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment.C0(SourceFile:1)
	at androidx.fragment.app.i.h1(SourceFile:1)
	at androidx.fragment.app.v.e(SourceFile:1)
	at androidx.fragment.app.v.m(SourceFile:1)
	at androidx.fragment.app.o.onCreateView(SourceFile:1)
	at androidx.fragment.app.l.n(SourceFile:1)
	at androidx.fragment.app.j.W(SourceFile:1)
	at androidx.fragment.app.j.onCreateView(SourceFile:1)
	at android.view.LayoutInflater.tryCreateView(Unknown Source:39)
	at android.view.LayoutInflater.createViewFromTag(Unknown Source:42)
	at android.view.LayoutInflater.createViewFromTag(Unknown Source:6)
	at android.view.LayoutInflater.rInflate(Unknown Source:92)
	at android.view.LayoutInflater.rInflateChildren(Unknown Source:9)
	at android.view.LayoutInflater.parseInclude(Unknown Source:302)
	at android.view.LayoutInflater.rInflate(Unknown Source:71)
	at android.view.LayoutInflater.rInflateChildren(Unknown Source:9)
	at android.view.LayoutInflater.parseInclude(Unknown Source:302)
	at android.view.LayoutInflater.rInflate(Unknown Source:71)
	at android.view.LayoutInflater.rInflateChildren(Unknown Source:9)
	at android.view.LayoutInflater.inflate(Unknown Source:112)
	at android.view.LayoutInflater.inflate(Unknown Source:29)
	at androidx.databinding.f.e(SourceFile:1)
	at androidx.databinding.g.m(SourceFile:1)
	at g3.b.v(SourceFile:1)
	at g3.b.u(SourceFile:1)
	at org.hollowbamboo.chordreader2.MainActivity.onCreate(Unknown Source:7)
	at android.app.Activity.performCreate(Unknown Source:127)
	at android.app.Activity.performCreate(Unknown Source:1)
	at android.app.Instrumentation.callActivityOnCreate(Unknown Source:3)
	at android.app.ActivityThread.performLaunchActivity(Unknown Source:627)
	at android.app.ActivityThread.handleLaunchActivity(Unknown Source:77)
	at android.app.servertransaction.LaunchActivityItem.execute(Unknown Source:79)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(Unknown Source:77)
	at android.app.servertransaction.TransactionExecutor.execute(Unknown Source:76)
	at android.app.ActivityThread$H.handleMessage(Unknown Source:172)
	at android.os.Handler.dispatchMessage(Unknown Source:19)
	at android.os.Looper.loopOnce(Unknown Source:182)
	at android.os.Looper.loop(Unknown Source:82)
	at android.app.ActivityThread.main(Unknown Source:123)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(Unknown Source:11)
	at com.android.internal.os.ZygoteInit.main(Unknown Source:312)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: org.hollowbamboo.chordreader2.chords.Chord
	at x0.x$l.a(SourceFile:1)
	at x0.t.d(SourceFile:1)
	at x0.t.f(SourceFile:1)
	at x0.t.a(SourceFile:1)
	at x0.t.a(SourceFile:1)
	... 48 more
Caused by: java.lang.ClassNotFoundException: org.hollowbamboo.chordreader2.chords.Chord
	at java.lang.Class.classForName(Native Method)
	at java.lang.Class.forName(Class.java:454)
	at java.lang.Class.forName(Class.java:379)
	... 53 more
Caused by: java.lang.ClassNotFoundException: org.hollowbamboo.chordreader2.chords.Chord
	... 56 more
```
Hope this helps troubleshoot the issue.
Cheers!

**Comment by TripleOh7** [comment]
Thanks for the crash log, I was trying to get a bug report but could not
find the save location

**Comment by galenthurber** [comment]
Same on Pie

**Comment by galenthurber** [comment]
How can I roll back or get older apo?

**Comment by TripleOh7** [comment]
> How can I roll back or get older apo?

Uninstall from f droid, scroll down and click install on an older version

**Comment by TripleOh7** [comment]
Your data should be saved because it should be saved to a different folder, although I would use a file manager to make sure before uninstalling so you don't delete your files.

**Comment by galenthurber** [comment]
How can I roll back or get older apk?

**Comment by NorfolkGreg** [comment]
I'm another installing over 2.2.0 via F-Droid on my Moto G62 5G that finds in crashes. (Had forgotten how I had managed to revert earlier and was left the dead app when I tried the upgrade a second time. Thanks to visiting here was reminded how. Now have 2.2.0 working again.)

**Comment by mikegilchrist** [comment]
I'm running into the same issues on a Pixel 6a running the latest CalyxOS.  Rolling back solves the problem. If you roll back on f-droid, don't forget to turn off auto updates for the app right after installing.

**Comment by Simx72** [comment]
Same issue 
Using android 12

**Comment by wtimme** [comment]
Running the latest state of `master` on an actual device does _not_ cause the app to crash, so I assume this crash is "introduced" when building the app for F-Droid.

Could the recent change regarding Multidex (cp. 3ddec21) be responsible for that, @AndInTheClouds?

**Comment by AndInTheClouds** [comment]
It seems like. But I build the app locally with debug signing, no crash. But when I build with release signing it crashes.
So I'm not sure if there may changed something by increasing the target API from 33 to 34. I'm still testing. I get app crash on all my devices from F-Droid build.

Traceback:
FATAL EXCEPTION: main
Process: org.hollowbamboo.chordreader2, PID: 6128
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.hollowbamboo.chordreader2/org.hollowbamboo.chordreader2.MainActivity}: android.view.InflateException: Binary XML file line #18 in org.hollowbamboo.chordreader2:layout/activity_main: Binary XML file line #17 in org.hollowbamboo.chordreader2:layout/content_main: Error inflating class fragment
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3993)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4173)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:114)
	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:231)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:152)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:93)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2595)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8592)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
Caused by: android.view.InflateException: Binary XML file line #18 in org.hollowbamboo.chordreader2:layout/activity_main: Binary XML file line #17 in org.hollowbamboo.chordreader2:layout/content_main: Error inflating class fragment
Caused by: android.view.InflateException: Binary XML file line #17 in org.hollowbamboo.chordreader2:layout/content_main: Error inflating class fragment
Caused by: java.lang.RuntimeException: Exception inflating org.hollowbamboo.chordreader2:navigation/mobile_navigation line 126
	at x0.t.b(SourceFile:1)
	at x0.j.o0(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment$b.g(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment$b.a(SourceFile:1)
	at d2.l.getValue(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment.a2(SourceFile:1)
	at androidx.navigation.fragment.NavHostFragment.C0(SourceFile:1)
	at androidx.fragment.app.i.h1(SourceFile:1)
	at androidx.fragment.app.v.e(SourceFile:1)
	at androidx.fragment.app.v.m(SourceFile:1)
	at androidx.fragment.app.o.onCreateView(SourceFile:1)
	at androidx.fragment.app.l.n(SourceFile:1)
	at androidx.fragment.app.j.W(SourceFile:1)
	at androidx.fragment.app.j.onCreateView(SourceFile:1)
	at android.view.LayoutInflater.tryCreateView(LayoutInflater.java:957)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:885)
	at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:849)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1011)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:972)
	at android.view.LayoutInflater.parseInclude(LayoutInflater.java:1146)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1007)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:972)
	at android.view.LayoutInflater.parseInclude(LayoutInflater.java:1146)
	at android.view.LayoutInflater.rInflate(LayoutInflater.java:1007)
	at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:972)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:570)
	at android.view.LayoutInflater.inflate(LayoutInflater.java:462)
	at androidx.databinding.f.e(SourceFile:1)
	at androidx.databinding.g.m(SourceFile:1)
	at g3.b.v(SourceFile:1)
	at g3.b.u(SourceFile:1)
	at org.hollowbamboo.chordreader2.MainActivity.onCreate(Unknown Source:7)
	at android.app.Activity.performCreate(Activity.java:8960)
	at android.app.Activity.performCreate(Activity.java:8938)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1526)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3975)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4173)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:114)
	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:231)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:152)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:93)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2595)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8592)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: org.hollowbamboo.chordreader2.chords.Chord
	at x0.x$l.a(SourceFile:1)
	at x0.t.d(SourceFile:1)
	at x0.t.f(SourceFile:1)
	at x0.t.a(SourceFile:1)
	at x0.t.a(SourceFile:1)
	... 49 more
Caused by: java.lang.ClassNotFoundException: org.hollowbamboo.chordreader2.chords.Chord
	at java.lang.Class.classForName(Native Method)
	at java.lang.Class.forName(Class.java:607)
	at java.lang.Class.forName(Class.java:512)
	... 54 more
Caused by: java.lang.ClassNotFoundException: org.hollowbamboo.chordreader2.chords.Chord
	... 57 more

Perhaps, has something to due with minifing sources/APK ?

Am 9. November 2024 12:07:37 MEZ schrieb wtimme ***@***.***>:

**Comment by AndInTheClouds** [comment]
fixed in version 2.3.2
